src/i_region.c

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright(C) 2025 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
```

Comment:
Could you add a header to describe what this file contains and what the code does? I would have problems finding words for this.

Reply:
"Wrapper for the virtual memory API, similar to the "region" crate in Rust." 

Why do we need it? We want the arena to grow dynamically. However, we can't use the simple `realloc` because it can allocate the memory buffer at a different address, which would invalidate all previously taken pointers. To guarantee that the memory is reallocated at the same address, we use a low-level API (`mmap` on POSIX and `VirtualAlloc` on Windows).

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/2349/commits/69420716d4c988e261e4c7e58b17bbc5ee6a6f6d

Reply:
I make it just "Restored key frame" in https://github.com/fabiangreffrath/woof/pull/2349/commits/34a0e3c0111558e644cf4d30f92996f634cc3ce9